sql/hive/src/main/scala/org/apache/spark/sql/hive/TableReader.scala

```diff
@@ -378,6 +387,9 @@ class HadoopTableReader(
   private def createNewHadoopRDD(partDesc: PartitionDesc, path: String): RDD[Writable] = {
     val newJobConf = new JobConf(hadoopConf)
     HadoopTableReader.initializeLocalJobConfFunc(path, partDesc.getTableDesc)(newJobConf)
+    // SPARK-36328: Reuse the FileSystem delegation token while querying partitioned hive table.
+    SparkHadoopUtil.get.addCurrentCredentials(newJobConf,
+      _broadcastedCredentials.value.value)
```

Comment:
wondering why we need this now?

Reply:
I'll delete it. I added it because considering that the new Hadoop API will make it easier to reuse tokens, but it violates the principle of minimizing modification points. @c21